File: packages/cli/templates/project/function-cron-trigger.ts
import type { InvocationContext, TimerEvent } from "@vibecloud/function-trigger-cron";

export async function {{HANDLER}}(
  event: TimerEvent,
  context: InvocationContext,
): Promise<void> {
  throw new Error(
    `cron handler is not implemented (${event.messages.length} messages, request ${context.requestId})`,
  );
}

File: packages/cli/templates/project/function-cron-trigger.go
package main

import (
	"context"
	"fmt"
)

type TimerEvent struct {
	Messages []struct {
		EventMetadata map[string]any `json:"event_metadata"`
		Details       map[string]any `json:"details"`
	} `json:"messages"`
}

func {{HANDLER}}(ctx context.Context, event TimerEvent) error {
	if err := ctx.Err(); err != nil {
		return err
	}
	return fmt.Errorf("cron handler is not implemented (%d messages)", len(event.Messages))
}
